Gems/EMotionFX/Code/Source/Editor/Plugins/SkeletonOutliner/JointPropertyWidget.h

```diff
@@ -39,10 +43,49 @@ namespace EMotionFX
 
     public slots:
         void Reset();
+
+    private slots:
+        void OnAddCollider(PhysicsSetup::ColliderConfigType configType, AZ::TypeId colliderType);
+        void OnAddToRagdoll();
+        void OnSearchTextChanged();
+
     private:
         AzToolsFramework::ReflectedPropertyEditor* m_propertyWidget{nullptr};
+        AddCollidersButton* m_addCollidersButton{nullptr};
+
+        ClothJointWidget* m_clothJointWidget{nullptr};
+        HitDetectionJointWidget* m_hitDetectionJointWidget{nullptr};
+        RagdollNodeWidget* m_ragdollJointWidget{nullptr};
+
         AZStd::unique_ptr<EMStudio::ActorInfo> m_actorInfo;
         AZStd::unique_ptr<EMStudio::NodeInfo> m_nodeInfo;
+
+        QLineEdit* m_filterEntityBox = nullptr;
+        AZStd::string m_filterString;
+    };
+
+
+    class AddCollidersButton : public QPushButton
+    {
+        Q_OBJECT
+    public:
+        AddCollidersButton(QWidget* parent=nullptr);
+        AZStd::string GetNameForColliderType(AZ::TypeId colliderType) const;
+    public slots:
+        void OnAddColliderActionTriggered(const QModelIndex& index);
+    signals:
+        void AddCollider(PhysicsSetup::ColliderConfigType configType, AZ::TypeId colliderType);
+        void AddToRagdoll();
+    protected slots:
+        void OnCreateContextMenu();
+    protected:
+        const AZStd::vector<AZ::TypeId> m_supportedColliderTypes = {
+            azrtti_typeid<Physics::BoxShapeConfiguration>(),
+            azrtti_typeid<Physics::CapsuleShapeConfiguration>(),
+            azrtti_typeid<Physics::SphereShapeConfiguration>()};
+
+        //QTreeView* m_treeView = nullptr;
+        QStandardItemModel* model = nullptr;
     };
 
-} // ns EMotionFX
+} // ns EmotionFX
```

Comment:
update: This I think should remain `EMotionFX` (also ns should really be `namespace` while we're here